rotkehlchen/externalapis/etherscan.py

```diff
@@ -240,9 +252,15 @@ def _query(
                     },
                 )
                 self.warning_given = True
-        else:
-            query_str += f'&apikey={api_key}'
 
+            if (raw_key := ROTKI_INCLUDED_KEYS.get(self.chain)) is not None:
+                api_key = ApiKey(raw_key)  # we warn right before that a key is needed
+                log.debug(f'Using default etherscan key for {self.chain}')
+            else:
+                self.msg_aggregator.add_error(msg := f'Etherscan for {self.chain} was queried but requires an API key. Please add one.')  # noqa: E501
```

Comment:
not sure I get this. What we need to achieve is:

1. If there is no api key use default. But warn user as we do now that they need an api key
2. If there is an api key from the user use that.